### PR TITLE
Changes due to applying patch from c lint

### LINKED ISSUE #84: Apply clinter patch
Extract patch from CI test and apply it, checking for any issues

### PR DESCRIPTION
**Pull request by jnothman**
Fix #84 

### All Submissions:

* [x] Have you followed the guidelines in our **CONTRIBUTING** document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

**Comment by jnothman** [comment]
C_linter now passes

File: src/hydromad-init.c [hunk 1]
```diff
@@ -1,52 +1,68 @@
...
+    {"sriv_system", (DL_FUNC)&sriv_system, 10},
+    {"swimp_core", (DL_FUNC)&swimp_core, 19},
+    {NULL, NULL, 0}};
 
 static const R_CallMethodDef CallEntries[] = {
-    {"_hydromad_simhyd_sim", (DL_FUNC) &_hydromad_simhyd_sim, 11},
-    {NULL, NULL, 0}
-};
+    {"_hydromad_simhyd_sim", (DL_FUNC)&_hydromad_simhyd_sim, 11},
+    {NULL, NULL, 0}};
```
Comment (josephguillaume) [inline src/hydromad-init.c:63]: In general I'd prefer to close the braces on its own line in a case like this

File: src/inversefilter.c [hunk 2]
```diff
@@ -2,157 +2,145 @@
...
+            }
+    */
+    // otherwise, do iterative bisection of U range to match c
+    while (fabs(Ubound0 - Ubound1) > 0.0001) {
+      // choose new U value between existing bounds
+      // TODO: use linear approximation rather than bisection
+      Utest = (Ubound0 + Ubound1) * 0.5;
+      ctest = a * pow(Utest, 1.0 + *lambda) + b * Utest;
+      Rprintf("U optim loop: Ut = %.3f, ct = %.3f, c = %.3f\n", Utest, ctest,
+              c);
```
Comment (josephguillaume) [inline src/inversefilter.c:123]: Leaving one variable on the next line is a little weird

File: src/riv.c [hunk 3]
```diff
@@ -2,135 +2,132 @@
...
+  xt = (double *)R_alloc(p, sizeof(double));
+  zt = (double *)R_alloc(p, sizeof(double));
+
+  // initialise outputs
+  for (i = 0; i < p * p; i++)
+    xz[i] = 0;
+  for (i = 0; i < p; i++)
+    xy[i] = 0;
+  for (i = 0; i < p * p; i++)
+    xx[i] = 0;
```
Comment (josephguillaume) [inline src/riv.c:41]: This is a case where the original is more readable, but the if statements elsewhere are probably better in this kind of format.

File: src/riv.c [hunk 4]
```diff
@@ -2,135 +2,132 @@
...
+  int m = order[1];
+  int p = n + m + 1;
+  double *xt;
+
+  xt = (double *)R_alloc(p, sizeof(double));
+
+  // initialise outputs
+  for (i = 0; i < p * p; i++)
+    xx[i] = 0;
+
```
Comment (josephguillaume) [inline src/riv.c:99]: Also more readable on a single line

**Comment by josephguillaume** [comment]
Thanks @jnothman ! I was really struggling with that.
I've added just a few comments on the file - do any of those seem like things at might be easy to change in the linter? Or should I just accept that I can't have it exactly as I would want :-P

**Comment by jnothman** [comment]
Automatic formatting (a la clang-format here, and stylr) is a bit controversial.

Last night we were discussing its adoption in https://github.com/scikit-learn/scikit-learn, and you land up balancing the aesthetic mediocrity of automatic formatting choices, with inconsistency and nitpicking that puts of new contributors.

You have a few options:
* wholesale adoption
* adopt some but not all of these changes as a once-off, and then switch to a linter that checks for issues but does not auto-format. You might not find any linter that allows if statements on one line by default, though.
* switch to a linter and change the code manually to conform
* disable linting C/++

**Comment by josephguillaume** [comment]
Thanks!
I'd prefer wholesale adoption but with some rules tweaked, but clearly that's not a priority right now.
I've opened https://github.com/josephguillaume/hydromad/issues/92 for a rainy day.

**Pull request merged**